src/test/java/com/epam/page/object/generator/integrationalTests/MainIntegrationTest.java

```diff
@@ -0,0 +1,283 @@
+
+        ValidatorsStarter validatorsStarter = new ValidatorsStarter(bc);
+        validatorsStarter.setCheckLocatorsUniqueness(checkLocatorUniqueness);
+
+        PageObjectsGenerator pog = new PageObjectsGenerator(parser, validatorsStarter,
+                javaPoetAdapter, RESOURCE_DIR, urls, PACKAGE_TEST_NAME);
+        pog.setForceGenerateFile(forceGenerateFiles);
+        return pog;
+    }
+
+    public MainIntegrationTest(List<TestClassDTO> list) throws MalformedURLException {
+        this.classesList = list;
+    }
+
+    @Parameters
+    public static Iterable<Object[]> data() throws IOException, ClassNotFoundException {
+
+        List<File> caseDirs = getMainDirsNames();
+        List<Object[]> testParameters = new ArrayList<>();
+
+        for (File caseDir : caseDirs) {
+            List<TestClassDTO> objectList = new ArrayList<>();
+            Properties caseProperties = new Properties();
+
+            List<File> insideDirs = (List<File>) FileUtils.listFilesAndDirs(
+                    caseDir,
+                    IO_FILE_FILTER,
+                    TrueFileFilter.INSTANCE
+            );
+            List<File> caseProps = (List<File>) FileUtils.listFiles(caseDir, null, false);
+            caseProperties.load(new FileInputStream(caseProps.get(0)));
+            insideDirs.remove(0);
+            for (File insideDir : insideDirs) {
+                filesList = (List<File>) FileUtils.listFiles(
+                        insideDir,
+                        TrueFileFilter.INSTANCE,
+                        TrueFileFilter.INSTANCE
+                );
+                for (File classFile : filesList) {
+                    String[] s = insideDir.getPath().split("[\\\\/]");
+                    TestClassDTO testClassDTO = new TestClassDTO(
+                            classFile.getName().split("\\.")[0],
+                            s[s.length - 2] + "/" + s[s.length - 1] + "/",
+                            caseProperties.getProperty("json"),
+                            caseProperties.getProperty("url"),
+                            Boolean.parseBoolean(caseProperties.getProperty("checkLocatorUniqueness")),
+                            Boolean.parseBoolean(caseProperties.getProperty("forceGenerateFiles"))
+                    );
+                    objectList.add(testClassDTO);
+                }
+
+            }
+            Object[] arr = new Object[]{objectList};
+            testParameters.add(arr);
+            filesList.clear();
+        }
+        return testParameters;
+    }
+
+    private static List<File> getMainDirsNames() {
+        List<File> caseDirs = (List<File>) FileUtils.listFilesAndDirs(
+                new File(RESOURCE_DIR + MANUAL_DIR),
+                IO_FILE_FILTER,
+                new IOFileFilter() {
+                    @Override
+                    public boolean accept(File file) {
+                        return Pattern.compile("manual([\\\\/])[a-zA-Z]+$").matcher(file.getPath())
```

Comment:
Same thing for regex: provide documentation. Also it would be good thing to move compilation of pattern out of iteration block.